File: src/ois_subjects/subjects.cs
// NOTE: it is recommended to use this even if you don't understand the following code.

using System;
using System.IO;
using System.Linq;
using System.Collections.Generic;

namespace subjects {

    class Program {
        static void Main(String[] args) {
            StreamReader streamReader = new StreamReader(Console.OpenStandardInput());
            StreamWriter streamWriter = new StreamWriter(Console.OpenStandardOutput());
            // uncomment the two following lines if you want to read/write from files
            // streamReader = new StreamReader("input.txt");
            // streamWriter = new StreamWriter("output.txt");

            IEnumerator<String> it = GetEnumerator(streamReader);

            int N = Convert.ToInt32(Next(it));
            int M = Convert.ToInt32(Next(it));

            int[] K = new int[N];
            int[,] S = new int[N, 5];
            for (int j = 0; j < N; ++j) {
                K[j] = Convert.ToInt32(Next(it));
                for (int i = 0; i < K[j]; ++i)
                    S[j, i] = Convert.ToInt32(Next(it));
            }

            int P = 0;
            int[,] ans = new int[1000000, 2];


            // INSERT YOUR CODE HERE


            streamWriter.WriteLine($"{P}");
            for (int j = 0; j < P; ++j) {
                for (int i = 0; i < 2; ++i)
                    streamWriter.Write($"{ans[j, i]} ");
                streamWriter.WriteLine($"");
            }

            streamReader.Close();
            streamWriter.Close();
        }

        static IEnumerator<String> GetEnumerator(StreamReader sr) {
            String line;
            while ((line = sr.ReadLine()) != null) {
                String[] tokens = line.Split(' ').Where(t => t.Length > 0).ToArray();
                foreach (String t in tokens) {
                    yield return t;
                }
            }
        }

        static String Next(IEnumerator<String> iterator) {
            return iterator.MoveNext() ? iterator.Current : null;
        }
    }
}

File: src/ois_subjects/subjects.py
#!/usr/bin/env python3
# NOTE: it is recommended to use this even if you don't understand the following code.

import sys

# uncomment the two following lines if you want to read/write from files
# sys.stdin = open('input.txt')
# sys.stdout = open('output.txt', 'w')

N, M = map(int, input().strip().split())

K = [0 for i in range(N)]
S = [[] for i in range(N)]
for j in range(N):
    S[j] = list(map(int, input().strip().split()))
    K[j], S[j] = S[j][0], S[j][1:]

P = 0
ans = [[0 for j in range(2)] for i in range(1000000)]


# INSERT YOUR CODE HERE


print(P)
for j in range(P):
    for i in range(2):
        print(ans[j][i], end=' ')
    print()

sys.stdout.close()
